src/plugins/validate-semantic/validators/oas3/refs.js

```diff
@@ -6,26 +6,58 @@ export const validateOAS3RefsForRequestBodiesReferenceRequestBodyPositions = ()
         const value = node.node
         const ref = value.$ref
 
-        if(!ref) {
+        if (!ref) {
           return acc
         }
 
         const [, refPath = ""] = ref.split("#")
```

Comment:
**change request:** use `refUrl` and `refPath` to feed the logic below this line, instead of using `ref` directly!

this gives us a higher-level view of what the ref actually is, which will make the logic below cleaner IMO. some usage examples:

```js
// a local ref
const [refUrl, refPath = ""] = "#/components/schemas/Pet".split("#")
console.log(refUrl) // ""
console.log(refPath) // "/components/schemas/Pet"

// a remote document ref
const [refUrl, refPath = ""] = "http://google.com/openapi.yaml".split("#")
console.log(refUrl) // "http://google.com/openapi.yaml"
console.log(refPath) // ""

// a remote path ref
const [refUrl, refPath = ""] = "http://google.com/openapi.yaml#/mySchema".split("#")
console.log(refUrl) // "http://google.com/openapi.yaml"
console.log(refPath) // "/mySchema"
```

```suggestion
        const [refUrl, refPath = ""] = ref.split("#")
```

Reply:
I specified some refPath, but as we are ignoring all external references. I don't know where to use refUrl.